### ISSUE TITLE
Snapshot testing support

### ISSUE THREAD

**Issue by sarod** [junit-team/junit5#1446]
## Overview

**Feature request.** 
When the method under test returns large text blocks it is useful to be able to store the text snapshot outside of the text class in a snapshot file.

This can be achieved outside of the test framework and can be done for Junit with something like with https://json-snapshot.github.io/

However having built-in support in the framework would allow for better developer experience.
This would allow command line test runners to interactively propose updating snapshot. Similary the IDEs could provide a better integration for updating the snapshots.

In the javascript world [Jest](https://facebook.github.io/jest/docs/en/snapshot-testing.html) is an example of test framework where snapshot is built-in. This allows a great developer experience to visualize and update snapshot using either the [command line runner ](https://facebook.github.io/jest/docs/en/snapshot-testing.html#updating-snapshots) or IDEs like webstorm ([see Jest Webstorm integration](https://www.jetbrains.com/help/webstorm/running-unit-tests-on-jest.html#ws_jest_snapshot_testing)).

**Comment by sormuras** [comment]
Neat idea and perfectly suited for an [extension](https://github.com/junit-team/junit5/wiki/Third-party-Extensions#junit-jupiter-extensions), methinks.

**Comment by sbrannen** [comment]
I agree with @sormuras: a feature like this should be implemented in a third-party extension.
